src/include/miopen/handle.hpp

```diff
@@ -211,6 +211,15 @@ struct MIOPEN_EXPORT Handle : miopenHandle
         return result;
     }
 
+    template <class V>
+    void ReadToVec(const Allocator::ManageDataPtr& ddata, V& output_vec)
+    {
+        using T = typename V::value_type;
+        assert(ddata);
+        assert(output_vec.size() > 0);
+        this->ReadTo(output_vec.data(), ddata, output_vec.size() * sizeof(T));
```

Comment:
Should we also check that `ddata` has at least `output_vec.size() * sizeof(T)` bytes allocated?

Reply:
Yes, I wanted that but today we don't store size of the buffer with the ManagedPtr, so I plan on doing this in the near future.